src/nvim/ex_cmds2.c

```diff
@@ -764,9 +764,13 @@ void ex_profile(exarg_T *eap)
     do_profiling = PROF_YES;
     profile_set_wait(profile_zero());
     set_vim_var_nr(VV_PROFILING, 1L);
-  } else if (do_profiling == PROF_NONE)
+  } else if (do_profiling == PROF_NONE) {
     EMSG(_("E750: First use \":profile start {fname}\""));
-  else if (STRCMP(eap->arg, "pause") == 0) {
+  } else if (STRCMP(eap->arg, "stop") == 0) {
+    profile_dump();
+    do_profiling = PROF_NONE;
+    set_vim_var_nr(VV_PROFILING, 0L);
```

Comment:
I am wondering what will happen if you do `profile start` after `profile stop`? It should start new profiling, but AFAIR neither of `profile_dump`, `set_vim_var` is clearing the associated structures.